deploy/modules/gcp/tidb-operator/main.tf

```diff
@@ -0,0 +1,156 @@
+
+  min_master_version = var.gke_version
+
+  lifecycle {
+    ignore_changes = [master_auth] // see above linked issue
+  }
+
+  maintenance_policy {
+    daily_maintenance_window {
+      start_time = "01:00"
+    }
+  }
+}
+
+resource "null_resource" "get-credentials" {
+  depends_on = [google_container_cluster.cluster]
+  provisioner "local-exec" {
+    command = "gcloud container clusters get-credentials ${google_container_cluster.cluster.name} --region ${var.gcp_region}"
+
+    environment = {
+      KUBECONFIG = var.kubeconfig_path
+    }
+  }
+
+  provisioner "local-exec" {
+    when = destroy
+
+    command = <<EOS
+kubectl get pvc -n tidb -o jsonpath='{.items[*].spec.volumeName}'|fmt -1 | xargs -I {} kubectl patch pv {} -p '{"spec":{"persistentVolumeReclaimPolicy":"Delete"}}'
+EOS
+
+
+    environment = {
+      KUBECONFIG = var.kubeconfig_path
+    }
+  }
+}
+
+data "local_file" "kubeconfig" {
+  depends_on = [google_container_cluster.cluster, null_resource.get-credentials]
+  filename = var.kubeconfig_path
+}
+
+resource "local_file" "kubeconfig" {
+  depends_on = [google_container_cluster.cluster, null_resource.get-credentials]
+  content = data.local_file.kubeconfig.content
+  filename = var.kubeconfig_path
+}
+
+provider "helm" {
+  alias = "initial"
+  insecure = true
+  # service_account = "tiller"
+  install_tiller = false # currently this doesn't work, so we install tiller in the local-exec provisioner. See https://github.com/terraform-providers/terraform-provider-helm/issues/148
+  kubernetes {
+    config_path = local_file.kubeconfig.filename
+  }
+}
+
+resource "null_resource" "setup-env" {
+  depends_on = [
+    google_container_cluster.cluster,
+    null_resource.get-credentials,
+    var.tidb_operator_registry,
+    var.tidb_operator_version,
+  ]
+
+  provisioner "local-exec" {
+    working_dir = path.cwd
+    interpreter = ["bash", "-c"]
+
+    command = <<EOS
+set -euo pipefail
+
+if ! kubectl get clusterrolebinding cluster-admin-binding 2>/dev/null; then
+  kubectl create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin --user $(gcloud config get-value account)
+fi
+
+if ! kubectl get serviceaccount -n kube-system tiller 2>/dev/null ; then
+  kubectl create serviceaccount --namespace kube-system tiller
+fi
+
+kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/${var.tidb_operator_version}/manifests/crd.yaml
+kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/${var.tidb_operator_version}/manifests/tiller-rbac.yaml
+kubectl apply -k manifests/local-ssd
+kubectl apply -f manifests/gke/persistent-disk.yaml
+
+helm init --service-account tiller --upgrade --wait
+until helm ls; do
+  echo "Wait until tiller is ready"
+  sleep 5
+done
+EOS
+
+
+    environment = {
+      KUBECONFIG = var.kubeconfig_path
+    }
+  }
+}
+
+data "helm_repository" "pingcap" {
+  provider   = "helm.initial"
+  depends_on = [null_resource.setup-env]
+  name       = "pingcap"
```

Comment:
As this is hardcoded here, the repo name is not necessary to be exposed in variables.tf.